src/query/block/column.go

```diff
@@ -223,10 +221,54 @@ func (cb ColumnBlockBuilder) AddCols(num int) error {
 	return nil
 }
 
+// PopulateColumns sets all columns to the given row size.
+func (cb ColumnBlockBuilder) PopulateColumns(size int) {
+	for i := range cb.block.columns {
+		cb.block.columns[i] = column{Values: make([]float64, size)}
```

Comment:
Worth working out the full size and doing a block alloc upfront?

i.e. `all := make([]float64, size*len(cb.block.columns))` and then subdividing for each one?  That would reduce num of individual allocations, we've seen good speed ups from bulk allocs.

Reply:
Neat trick!